MLS.Agent/NativeAssemblyLoadHelper.cs

```diff
@@ -49,6 +54,119 @@ public void Handle(FileInfo assemblyFile)
             }
         }
 
+        private IEnumerable<string> ProbingFilenames(string name)
```

Comment:
the name passed in here is the value specified with the DllAttribute.  by convention, the dll may have:
1. A full path
2. A relativepath
3. lib prepended
4. on windows .dll or .exe as a suffix.
5. on linux .so as a suffix
6. on macos .dynlib as a suffix

This function enumerates the possibilities.